eng/build.sh

```diff
@@ -214,7 +214,8 @@ function Test() {
   projectname=$(basename -- "$testproject")
   projectname="${projectname%.*}"
   testlogpath="$artifacts_dir/TestResults/$configuration/${projectname}_$targetframework.xml"
-  args="test \"$testproject\" --no-restore --no-build -c $configuration -f $targetframework --test-adapter-path . --logger \"xunit;LogFilePath=$testlogpath\" --blame --results-directory $artifacts_dir/TestResults/$configuration -p:vstestusemsbuildoutput=false"
+  args="test \"$testproject\" --no-restore --no-build -c $configuration -f $targetframework --test-adapter-path . --logger \"xunit;LogFilePath=$testlogpath\" --blame-hang-timeout 5minutes --results-directory $artifacts_dir/TestResults/$configuration -p:vstestusemsbuildoutput=false"
+  args+=" -- xUnit.MaxParallelThreads=4"
```

Comment:
Sooo it looks like `build.ps1` is doing something else on this topic, but regardless - shouldn't this be driven by xunit.runner.json files?

Reply:
It would be ideal, but this throttling is just for the sake of the CI. Locally it runs ok full speed.

Reply:
To add, this was tuned purely by experiment, IIRC without this throttling CI on Linux or Mac was running out of memory. The reason it is different in Build.ps1 is that I had easier time working with powershell on Windows. I'm frankly afraid of touching this bash script 😄

Reply:
Alright, no I mean if you want to keep this, maybe just add some comment - tho I think it might have become redundant?

Either way, don't fear 💪